### PR TITLE
Support ccache and done some cleanup

### PR DESCRIPTION
**Pull request by Thaodan**
Support using of ccache and cleaned some parts of the program

### CODE REVIEW

File: common/clean-chroot-manager64.in [hunk 1]
```diff
@@ -221,9 +226,15 @@ build() {
...
+        
         [[ -n "$RUNNAMCAP" ]] && RUNNAMCAP=-n
+        [[ -n "$USE_CCACHE" ]] && { USE_CCACHE=-d$CCACHE_DIR
+                                    local ccache_dir="CCACHE_DIR=$CCACHE_DIR" }
         
-	PACKAGER=$PACKAGER MAKEFLAGS=-j$THREADS nice -19 /usr/bin/makechrootpkg -c -u $RUNNAMCAP -r "$CHROOTPATH64"
+  	PACKAGER=$PACKAGER MAKEFLAGS=-j$THREADS nice -19 /usr/bin/makechrootpkg $USE_CCACHE \
+                -c -u $RUNNAMCAP \
+                -r "$CHROOTPATH64" \
+                -- $ccache_dir
```
Comment (eli-schwartz) [inline common/clean-chroot-manager64.in]: Shouldn't you just ask for a CCACHE_DIR and then bind-mount it directly as:
 `-d "$CCACHE_DIR:/build/.ccache"`?

Keeping in mind also that, regardless, user-supplied paths can contain spaces which will break your PR.
I suggest assembling the extra arguments via a (properly-quoted) array.
You should be able to unify build() and buildnc() too.

**Comment by Thaodan** [comment]
What you mean with ask?
I need to set CCACHE_DIR in the chroot and tell makepkg to use it this rhe 
reason why I added it as arg to.
But the  thing with the array better that is true.

**Comment by eli-schwartz** [comment]
By ask, I meant via the configuration...

Point being, once you have a CCACHE_DIR, you can bind-mount that **to where ccache expects it** instead of messing with the environment variable. I even gave the necessary makechrootpkg parameter.

There is no need to set an environment variable if ccache is using the default location!

**Comment by graysky2** [comment]
@Thaodan - please rebase the PR

**Comment by Thaodan** [comment]
sorry but whats the problem that I've made more than one commit?

**Comment by graysky2** [comment]
My brain ... doing too many things today and would like to see the changes in a single page :p 

If you have 3 commits just:
`git rebase -i @~3`
Leave 'pick' selected for the first line and change the next to instances to 'squash'
save
then `git push origin +master` or whatever branch. Easy.

**Comment by graysky2** [comment]
Ping?

**Pull request closed without merging**